### PR TITLE
CDAP-4340 Documentation for accessing MapReduce counters from the WorkflowToken

### PR DESCRIPTION
**Pull request by awholegunch**
Re-creation of PR https://github.com/caskdata/cdap/pull/4664

Fixes errors in the WikipediaPipeline Example found in testing.

Fix for: https://issues.cask.co/browse/CDAP-4340

Passes [Quick Build](http://builds.cask.co/browse/CDAP-DOB74-3) (#3)
[Full Doc Build](http://builds.cask.co/browse/CDAP-DDB27-3) (#3)

[Full Build](http://builds.cask.co/browse/CDAP-DUT3325-4) (#4)

Page of Interest: 
- [Developers' Manual: Workflow](http://builds.cask.co/artifact/CDAP-DOB74/shared/build-3/Docs-HTML/3.3.0-SNAPSHOT/en/developers-manual/building-blocks/workflows.html#workflow-tokens)
- [Examples: Wikipedia Pipeline](http://builds.cask.co/artifact/CDAP-DOB74/shared/build-3/Docs-HTML/3.3.0-SNAPSHOT/en/examples-manual/examples/wikipedia-data-pipeline.html)

### CODE REVIEW

**Comment by anwar6953** [comment]
I think this is a valid comment that isn't addressed:
```
Is this complete? The Jira explicitly mentions to document:

How to read the MR counters from the WorkflowToken
How to read the custom counters from the WorkflowToken
I don't see 2. as part of the PR
```
Source: https://github.com/caskdata/cdap/pull/4664#issuecomment-162710791

**Comment by awholegunch** [comment]
I've updated the PR with these changes:
- Added an example from the WikipediaPipeline example, showing setting and getting a workplace token in the user scope
- To make the code sample clearer, I used "emphasize-lines". In order to get a color that wasn't "ugly-yellow", I modified the CSS for the page, only to discover that we load the ``pygments.css`` after our custom theme's CSS, making it impossible to alter. I flipped the order of loading so that we can easily change the coloring of code without having to change the pygments file.
- As I modified the example (making the two lines usage of ONLINE_MODE identical, to clarify the usage), I have rebuilt and will retest the example.
- Added the example file as a guarded file to the build script, and cleaned up a couple of minor nits.

**Comment by awholegunch** [comment]
When I was testing the example, I found an error in the example code. I fixed that and add some clarifying language and formatting changes.

**Comment by anwar6953** [comment]
I'm not sure that's what Andreas meant in his comment - https://github.com/caskdata/cdap/pull/4664#issuecomment-162710791.
@anew , can you clarify?

**Comment by anew** [comment]
@anwar6953 I cannot see where you made that last comment, and what it's related to. My comment was to say that this PR needs more content.

**Comment by awholegunch** [comment]
@anew @anwar6953 Andrea's comment (from above):
```
Is this complete? The Jira explicitly mentions to document:

How to read the MR counters from the WorkflowToken
How to read the custom counters from the WorkflowToken
I don't see 2. as part of the PR
```
I addressed #2 (How to read the custom counters from the WorkflowToken) by adding a code sample from the WikipediaPipeline Example. (Line 308, https://github.com/caskdata/cdap/pull/4731/files#diff-8a7dca9cdf4aed2b447969eb6221ebdc). You can see the inclusion on [the rendered page](http://builds.cask.co/artifact/CDAP-DOB74/shared/build-3/Docs-HTML/3.3.0-SNAPSHOT/en/developers-manual/building-blocks/workflows.html#workflow-tokens); it's the second of the two code samples:

```
token.put(ONLINE_MODE, Value.of(runtimeArguments.get(MODE_KEY).equalsIgnoreCase(ONLINE_MODE)));
...
Value online = token.get(ONLINE_MODE, EnoughDataToProceed.class.getSimpleName(), WorkflowToken.Scope.USER);
```
Is that sufficient? They aren't "counters" per se; they are passing a token with a value, but I thought a developer would understand that this could be generalized.

**Comment by anew** [comment]
I think what is missing (and what was misleading for the originator of this Jira) is a way to access custom MR counters (that are not in the org.apache.hadoop.mapreduce.* groups).

**Comment by anew** [comment]
The mistake is actually in the "Conditional Node" section, which uses the wrong group name to access the custom counter. Overall, I think that the examples here are a little hard to follow. I will work with @sagarkapare when he is back from his break to improve this doc.

**Comment by anew** [comment]
Putting this on hold until @sagarkapare is back.
